File: src/runtime/boxes/bb_succeed.c
/* _XSUCF    SUCCEED     always γ zero-width; outer loop retries */
#include "bb_box.h"
#include <stdlib.h>
#include <string.h>


spec_t bb_succeed(void *zeta, int entry)
{
    (void)zeta; (void)entry;
    return spec(Σ+Δ,0);
}

succeed_t *bb_succeed_new(void)
{ return calloc(1,sizeof(succeed_t)); }

File: src/runtime/boxes/BbSucceed.java
/**
 * BbSucceed.java — SUCCEED: always γ zero-width; outer scan loop retries
 * Port of bb_succeed.c / bb_succeed.s
 */
class BbSucceed extends BbBox {

    public BbSucceed(MatchState ms) { super(ms); }

    @Override public Spec alpha() { return new Spec(ms.delta, 0); }
    @Override public Spec beta()  { return new Spec(ms.delta, 0); }
}
